botorch/acquisition/utils.py

```diff
@@ -61,8 +62,13 @@ def get_acquisition_function(
         constraints: A list of callables, each mapping a Tensor of dimension
             `sample_shape x batch-shape x q x m` to a Tensor of dimension
             `sample_shape x batch-shape x q`, where negative values imply
-            feasibility. Used when constraint_transforms are not passed
-            as part of the objective.
+            feasibility. Used only for qEHVI and qNEHVI.
```

Comment:
@sdaulton, @saitcakmak I guess this special handling is needed b/c we don't yet have posterior transforms implemented for MOO? Do we have any concrete plans to change this? It would be nice to make this interface more consistent.

Reply:
Hmm, I thought the reason is that in `qNEHVI` and `qEHVI` the constraints are defined directly in the acquisition function whereas in the other acqfs, we define the constraints as part of the `objective` and then give the objective to the constructor of the acquisition function. Or is there another reason?

Reply:
I think @jduerholt is correct here. IIRC, in q(N)EHVI we use constraints to weight the HVI, whereas in other acqfs, we feasibility weight the samples directly (via the objective).